theseus/geometry/functional/lie_group.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+import abc
+
+from typing import List, Optional, Tuple
+from .utils import check_jacobians_list
+
+
+class ExpMap(torch.autograd.Function):
+    @classmethod
+    @abc.abstractmethod
+    def forward(cls, ctx, tangent_vector):
+        pass
+
+
+def ExpMapFactory(module):
```

Comment:
Why not pass the two functions directly? `exp_autograd_fn` and `jexp_impl`? No need to be fancy here.